modules/ROOT/pages/sample-data-java-sdk.adoc

```diff
@@ -0,0 +1,472 @@
+= Sample Data
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Sample data enables operations and sources to provide examples of the output
+they produce. These examples show information obtained from a live system
+through a connection or the output format from a connectionless example.
+
+Sample data is supported by Mule 4.4.0 or later, however, all Anypoint Connectors
+can use this feature, regardless of their Mule version.
+
+== SDK APIs
+
+=== SampleDataResolver
+
+Implement this API to enhance operations and message sources with sample data
+that is representative of the data the component outputs in real production
+scenarios.
+
+Implementations must:
+
+* Provide a default constructor.
+* Provide samples for both the payload and the message attributes, which
+return a Result<T,A> object. Generic types always match the return types of
+the enhanced component. For components that return dynamic types, each
+implementation returns instances of the correct dynamic type. If the
+operation does not return attributes, use Result<T, Void>.
+* Guarantee that fetching sample data does not generate any significant side
+effects on the target system.
+* Not be used to return large objects or data streams. The data should fit into
+the memory of a small worker instance.
+
+Implementations can access parameters, configurations, and connections to
+generate the sample data through the `@Parameter`, `@Config`, and `@Connection`
+annotations.
+
+For connectors, implementations are the most valuable when returning real data
+from the system they connect to instead of returning hard coded values.
+
+The following example illustrates a SampleDataResolver API:
+
+[source,java,linenums]
+----
+/**
+* [...]
+*
+* @param <T> the generic type of the output value
+* @param <A> the generic type of the message attributes
+* @since 1.4
+*/
+@MinMuleVersion("4.4")
+public interface SampleDataProvider<T, A> {
+
+ /**
+  * An ID that identifies each implementation of this interface. It must be unique among implementations in the same
+  * extension. Subsequent invocations must always return the same fixed value.
+  *
+  * @return the resolver's id
+  */
+ String getId();
+
+ /**
+  * Obtains and returns the sample information.
+  *
+  * @return a {@link Result} object
+  * @throws SampleDataException if the sample cannot be obtained
+  */
+ Result<T, A> getSample() throws SampleDataException;
+}
+----
+
+Streaming is not supported for:
+
+* Operations or sources that return an InputStream. The resolver always returns
+implementations that do not require a connection, such as a ByteArrayInputStream
+or any other offline implementations that have already procured the entire data
+stream before returning the getSample() method, as the InputStream is fully
```

Comment:
GLOBALLY replace "as" with "because" when that is what you mean, for translatability:
```suggestion
stream before returning the getSample() method, because the InputStream value is fully
```

Reply:
I'm not sure that "value" is the correct noun object, but the name/syntax InputStream and similar needs to be presented preceded by "the" and followed by a noun object, or without either the "the" or object.
GLOBAL

Reply:
Committed this and globally changed